File: lib/scoring.ts
export function calculateRelativeToPar(strokes: number, par: number) {
  return strokes - par
}

export function calculateRoundTotals<T extends { par: number; strokes: number }>(holes: T[]) {
  const totalPar = holes.reduce((sum, hole) => sum + hole.par, 0)
  const totalStrokes = holes.reduce((sum, hole) => sum + hole.strokes, 0)
  const relativeToPar = totalStrokes - totalPar

  return {
    totalPar,
    totalStrokes,
    relativeToPar,
  }
}

export function formatRelativeToPar(value: number) {
  if (value === 0) return 'E'
  return value > 0 ? `+${value}` : `${value}`
}

File: rounds/scoring.py
def calculate_round_totals(holes):
    """holes: iterable of objects with .par and .strokes attributes."""
    total_par = sum(h.par for h in holes)
    total_strokes = sum(h.strokes for h in holes)
    return {
        "total_par": total_par,
        "total_strokes": total_strokes,
        "relative_to_par": total_strokes - total_par,
    }


def format_relative_to_par(value: int) -> str:
    if value == 0:
        return "E"
    return f"+{value}" if value > 0 else str(value)
